SH4RE/SH4RE/Search/FilterSheetView.swift

```diff
@@ -0,0 +1,134 @@
+                        })
+                    )
+                )
+        }
+    }
+}
+
+struct FilterSheetView: View {
+    
+    var dropDownList = ["Film & Photography", "Audio Visual Equipment", "Projectors & Screens", "Drones", "DJ Equipment", "Transport", "Storage", "Electronics", "Party & Events", "Sports", "Musical Instruments", "Home, Office & Garden", "Holiday & Travel", "Clothing"]
+    @State private var dropDownSelection: String = ""
+
+    @State private var location: String = ""
+    @State private var minPrice: String = ""
+    @State private var maxPrice: String = ""
+    @State private var maxDistance: String = ""
+    
+    @Binding var showingFilterSheet: Bool
+    
+    @State private var minRating = 0.0
+
+    fileprivate func NumericTextField(label: String, textEntry: Binding<String>) -> some View {
+        return TextField(label, text: textEntry)
+            .keyboardType(.numberPad)
+            .textFieldStyle(textInputStyle())
+            .onReceive(Just(textEntry.wrappedValue)) { newValue in
+                let filtered = newValue.filter { "0123456789".contains($0) }
+                if filtered != newValue {
+                    textEntry.wrappedValue = filtered
+                }
+            }
+    }
+    
+    var body: some View {
+        ScrollView {
+            VStack (alignment: .leading){
+                Group {
+                    HStack {
+                        Text("Filter")
+                            .font(.title)
+                            .bold()
+                        Spacer()
+                        Button("Apply", action: {
+                            showingFilterSheet.toggle()
+                        }).buttonStyle(primaryButtonStyle(width: 80))
+                        
+                    }
+                    
+                    //Location
+                    VStack (alignment: .leading) {
+                        Text("Location")
+                            .font(.title2)
+                        LocationEntryField(location: $location)
+                    }
+                    
+                    //Categories
+                    VStack (alignment: .leading){
+                        Text("Category")
+                            .font(.title2)
+                        DropdownMenu(label: "Categories", options: dropDownList, selection: $dropDownSelection, useClear: true, clearValue: "")
+                    }
+                    
+                    //Price
+                    VStack (alignment: .leading) {
+                        Text("Daily Price")
+                            .font(.title2)
+                        HStack {
+                            NumericTextField(label: "Min", textEntry: $minPrice)
+                            NumericTextField(label: "Max", textEntry: $maxPrice)
+                        }
+                    }
+
+                    //Distance
+                    VStack (alignment: .leading) {
```

Comment:
I think the max distance filter should be moved below the location selector since they are related

Reply:
fair enough, i felt like tha too, howevver it then moved the "price" filter off the screen (until you pull the filter sheet up higher) and I felt like people would use price more than distance. But yeah I was unsure so I will change it